### PR TITLE
Add endpoint for CANFundingBudgets

### PR DESCRIPTION
**Pull request by rajohnson90**
## What changed

This PR adds the endpoints for /can-funding-budgets/ and allows the creation, update, and deletion of those objects within our database. In addition

## Issue

[OPS-2837](https://github.com/HHS/OPRE-OPS/issues/2837)

## How to test
Run backend python tests. There are no front end tests to run because this endpoint is not being consumed yet until we add the funding budget workflow. 

## Definition of Done Checklist
- [x] OESA: Code refactored for clarity
- [x] OESA: Dependency rules followed
- [x] Automated unit tests updated and passed
- [x] Automated integration tests updated and passed
- [x] Automated quality tests updated and passed
- [x] Automated load tests updated and passed
- [x] Automated a11y tests updated and passed
- [x] Automated security tests updated and passed
- [x] 90%+ Code coverage achieved
- [x] Form validations updated

### CODE REVIEW

File: frontend/src/pages/agreements/approve/ApproveAgreement.jsx
Comment (fpigeonjr) [inline frontend/src/pages/agreements/approve/ApproveAgreement.jsx:138]: did we mean to leave in Debug Code component?

File: backend/ops_api/ops/schemas/cans.py
Comment (fpigeonjr) [inline backend/ops_api/ops/schemas/cans.py:98]: should we update the `openapi` schema as well?

**Review by fpigeonjr — COMMENTED** [review]
🧪 tests pass but will defer to others

File: frontend/src/pages/agreements/approve/ApproveAgreement.jsx
Comment (rajohnson90) [reply frontend/src/pages/agreements/approve/ApproveAgreement.jsx:138]: I did not. A bunch of changes came into the ApproveAgreement.jsx when I merged main in. I haven't touched any front end files myself, aside to fix the things that were brought in by the merge. Really unclear why some of that work was able to be merged into main while failing the linter! I'll remove this.

Comment (rajohnson90) [reply frontend/src/pages/agreements/approve/ApproveAgreement.jsx]: Removed

File: backend/ops_api/ops/schemas/cans.py
Comment (rajohnson90) [reply backend/ops_api/ops/schemas/cans.py:98]: Yes this is a good point. I definitely should update the openapi and add the schema and endpoints.

Comment (rajohnson90) [reply backend/ops_api/ops/schemas/cans.py:97]: Done in my latest changelist.

**Review by fpigeonjr — CHANGES_REQUESTED** [review]
image

**Comment by rajohnson90** [comment]
I can't respond to Frank directly for some reason, but I imagine that will be fixed once your change is merged in.

**Comment by fpigeonjr** [comment]
merged in #2846 to `main` @rajohnson90 so it should resolve the openApi issue

**Review by maiyerlee — APPROVED** [review]

**Review by johndeange — APPROVED** [review]

**Pull request merged**
